src/devices/cpu/z80/tmpz84c015.h

```diff
@@ -26,8 +26,11 @@ class tmpz84c015_device : public z80_device
 {
 public:
 	tmpz84c015_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t);
+	tmpz84c015_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t, address_map_constructor io_map);
 
 	// configuration helpers
+	template <int Channel> void set_clk_trg(u32 clock) { if (m_ctc) m_ctc->set_clk<Channel>(clock); else subdevice<z80ctc_device>(m_ctc.finder_tag())->set_clk<Channel>(clock); }
+	template <int Channel> void set_clk_trg(const XTAL &xtal) { if (m_ctc) m_ctc->set_clk<Channel>(xtal); else subdevice<z80ctc_device>(m_ctc.finder_tag())->set_clk<Channel>(xtal); }
```

Comment:
Is there really any point having the `if` statement here?  The primary case for calling this is during configuration of the parent device where `m_ctc` will be unresolved.  You basically need to call `subdevice`.

Reply:
I don't have strong opinion here. That was @smf- suggestion.

Reply:
Just use the `subdevice` calls – it’ll make the line shorter and easier to read.